arbor/simulation.cpp

```diff
@@ -261,6 +265,14 @@ simulation_state::simulation_state(
     epoch_.reset();
 }
 
+void simulation_state::update_connections(const recipe& rec) {
+    auto spikes = communicator_.num_spikes(); // TODO(TH) add those to the next communicator
+    communicator_ = arb::communicator(rec, ddc_, source_resolution_map_, target_resolution_map_, *ctx_);
+    // Use half minimum delay of the network for max integration interval.
+    t_interval_ = communicator_.min_delay()/2;
```

Comment:
Because we use the `min_delay` in our time stepping and it is defined as `min { c.delay | c in connections }`.
Thus an update might change the minimum.